pkg/scheduler/factory/plugins.go

```diff
@@ -101,6 +101,60 @@ type AlgorithmProviderConfig struct {
 	PriorityFunctionKeys sets.String
 }
 
+// Snapshot is used to store current state of registered predicates and priorities.
+type Snapshot struct {
+	fitPredicateMap        map[string]FitPredicateFactory
+	mandatoryFitPredicates sets.String
+	priorityFunctionMap    map[string]PriorityConfigFactory
+	algorithmProviderMap   map[string]AlgorithmProviderConfig
+}
+
+// Copy returns a snapshot of current registered predicates and priorities.
+func Copy() *Snapshot {
+	schedulerFactoryMutex.RLock()
+	defer schedulerFactoryMutex.RUnlock()
+
+	copy := Snapshot{
+		fitPredicateMap:        make(map[string]FitPredicateFactory),
+		mandatoryFitPredicates: sets.NewString(),
+		priorityFunctionMap:    make(map[string]PriorityConfigFactory),
+		algorithmProviderMap:   make(map[string]AlgorithmProviderConfig),
+	}
+	for k, v := range fitPredicateMap {
+		copy.fitPredicateMap[k] = v
+	}
+	for k := range mandatoryFitPredicates {
+		copy.mandatoryFitPredicates[k] = struct{}{}
```

Comment:
We could use `copy.mandatoryFitPredicates.Insert(k)` here.

Reply:
Not that necessary, IMO it's clear enough. Esp. we can save a function call.

Reply:
I also find insert nicer to use here instead of the awkward "struct{}{}". I don't think you are saving a function call here, "copy.mandatoryFitPredicates[k] = struct{}{}" is basically a call to the "operator=" function, I would be very surprised if there is any difference between insert and operator=, changing it to insert it is up to you though.